packages/itwinui-react/src/core/Overlay/Overlay.tsx

```diff
@@ -0,0 +1,101 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import {
+  polymorphic,
+  type PolymorphicForwardRefComponent,
+} from '../utils/index.js';
+import cx from 'classnames';
+
+export type OverlayComponentProps = {
+  /**
+   * The sub-component elements of Overlay.
+   */
+  children?: React.ReactNode;
+  /**
+   * Placeholder for Progress Indicator in Overlay
+   */
+  content?: React.ReactNode;
+};
+
+// --------------------------------------------------------------------------------
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const OverlayComponent = React.forwardRef((props, forwardedRef) => {
+  const { content, children, ...rest } = props;
+
+  return (
+    <OverlayWrapper {...rest}>
+      <OverlayMessage>{content}</OverlayMessage>
+      <OverlayHiddenContent>{children}</OverlayHiddenContent>
+    </OverlayWrapper>
+  );
+}) as PolymorphicForwardRefComponent<'div', OverlayComponentProps>;
+OverlayComponent.displayName = 'Overlay';
+
+// --------------------------------------------------------------------------------
+
+const OverlayHiddenContent = React.forwardRef((props, ref) => {
+  const { as: Box = 'div', children, ...rest } = props;
```

Comment:
`as: Box = 'div'` is incorrect. You need to import the `Box` component and use that